### PR TITLE
implement Neuromag SSP similar to montage/balancing for CTF synthetic gradients , see issue #1952

### PR DESCRIPTION
**Pull request by ualsbombe**
So here's a first attempt. Check the test script - I can't plot the multiplot- maybe because the units are "unknown"

I have only run it on a timelocked so far

`denoised_timelock.grad.chanunit`

### CODE REVIEW

**Review by gdemarchi — CHANGES_REQUESTED** [review]
Just s useless comment :-) Thanks for putting me in, just fix my name (line 26) as long as you are in time, I'm not yet brasilian ... GianpaOlo Demarchi, just that :-)

File: ft_denoise_ssp.m [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+%   cfg.inputfile   =  ...
+%   cfg.outputfile  =  ...
+% If you specify one of these (or both) the input data will be read from a *.mat
+% file on disk and/or the output data will be written to a *.mat file. These mat
+% files should contain only a single variable, corresponding with the
+% input/output structure.
+%
+% See also FT_PREPROCESSING, FT_DENOISE_PCA
+
+% Copyright (C) 2004-2022, whose names to put, Gianpaulo Demarchi, Lau
```
Comment (gdemarchi) [inline ft_denoise_ssp.m:26]: Gianpaolo Demarchi instead

**Comment by ualsbombe** [comment]
So sorry, Gianpaolo - I try my best not to butcher names, but this one slipped. I did not want to imply that you were Brazilian ;-)

**Comment by schoffelen** [comment]
Shoot, I messed it up (again) by pulling the PR into a more recent version of FT than the one from which Lau created this PR, causing more changed files than needed. I will look into this. My intention was to update the test script into a function, and made a half-baked attempt to write in some code that allows access to ftp-data from outside the Donders (@robertoostenveld this is not related to this PR per se, but might be considered to be a trick to allow running of the test functions from outside Donders). This being said, the fif-file I used from the natmeg workshop does not seem to contain SSP-projectors, so it's unusable for now. Ergo, the whole exercise on this rainy Saturday afternoon was for nothing, given that I have messed it up now. Will fix later.

Thanks Jean-Paul and Lau so far!

**Comment by schoffelen** [comment]
Sorry for the mess, I created a new PR #1988 with only the 3 original files changed. I suggest to close this one, and merge the other one, once we agree that all is in order

**Comment by gdemarchi** [comment]
> functions from outside Donders). This being said, the fif-file I used from the natmeg workshop does not seem to contain SSP-projectors, so it's unusable for now. Ergo, the whole exercise on this rainy Saturday afternoon was for nothing, given that I have messed it up now. Will fix later.

Do you want an ethics-less .fif file to test upon, like a small empty room measurement? We're plenty of, and this would suffice given the fact that SSP projectors (or at least the Neuromag/Elekta/MEGIN ones) are indeed built on empty room ... no brain there, but would be enough to see whether a) the function works and b) the projectors make the spectrum cleaner ...  just let me know (so I would feel more useful in this not-so-rainy Saturday of (my own) baby sitting 😄 )

**Pull request closed without merging**